bin/warmup_stats

```diff
@@ -231,22 +234,62 @@ def check_environment(need_outliers=True, need_changepoints=True, need_latex=Tru
 class BenchmarkFile(object):
     """Information about each benchmark, including CLI options to other scripts."""
 
-    def __init__(self, csv_filename, options, python_path, pypy_path, pdflatex_path, r_path):
-        self.basename = os.path.splitext(csv_filename)[0]
-        self.csv_filename = csv_filename
+    def __init__(self, filename, options, python_path, pypy_path, pdflatex_path, r_path):
+        self.basename = os.path.splitext(filename)[0]
         self.language = options.language
         self.vm = options.vm
         self.uname = options.uname
         self.python_path = python_path
         self.pypy_path = pypy_path
         self.pdflatex_path = pdflatex_path
         self.r_path = r_path
+        self.csv_filename = None
+        self.krun_filename = None
+        self.krun_filename_outliers = None
+        self.krun_filename_changepoints = None
+        assert filename.endswith('.csv') or filename.endswith('.json.bz2'), \
+            'Unknown file type: %s. Please use CSV or Krun output.' % filename
+        if filename.endswith('.csv'):
+            self.csv_filename = filename
+            # self.iterations is set in self.convert_to_krun_json().
+        if filename.endswith('_changepoints.json.bz2'):
+            self.krun_filename_changepoints = filename
+        if '_outliers_w' in filename:
+            self.krun_filename_outliers = filename
+        if filename.endswith('.json.bz2'):
+            self.krun_filename = filename
+            if not os.path.exists(filename):
+                fatal('File %s does not exist.' % filename)
+            # If the input file comes from Krun, we must still set
+            # self.iterations, which is needed by other methods. We assume the
+            # same number of iterations for all pexecs.
+            data = read_krun_results_file(filename)
+            try:
+                for bench in data['wallclock_times']:
+                    for pexec in data['wallclock_times'][bench]:
+                        if pexec == []:  # Crashed benchmark.
+                            continue
+                        else:
+                            self.iterations = len(pexec)
+                            debug('%d iterations per pexec in %s.' % (self.iterations, filename))
+                            raise StopIteration()
```

Comment:
Has this been tested? Won't the inner `for` loop catch this and not the outer `try`?

Reply:
It has been tested and it thinks the test files from @fsfod have 1500 iterations, which I believe is correct. If I put a print statement in the `except` block it fires. The code was adapted from the plotting script BTW.

Reply:
Ah, I guess the `StopIteration` only does "bad" things if it's in the `for` expression.

That said, I don't like this idiom, which is a poor-man's goto. I'd much prefer we used explicit booleans. [I bet I'm not the only person who wonders what catching `StopIteration` does!] Up to you whether you fix it now or later.